include/bluetooth/gatt.h

```diff
@@ -1773,6 +1773,38 @@ int bt_gatt_unsubscribe(struct bt_conn *conn,
  */
 void bt_gatt_cancel(struct bt_conn *conn, void *params);
 
+/** @brief Encode GATT UUID of current services into a bt_data struct.
+ *
+ * The data will include both primary and secondary service UUIDs.
+ * Mandatory services such as the GATT and GAP services will not be included.
+ *
+ * @param[in] type
+ * Type of the UUIDs. Shall be one of following:
+ *  - BT_DATA_UUID16_SOME:  Encode 16-bit UUIDs until @p data_buf is full.
+ *  - BT_DATA_UUID16_ALL:   Encode all 16-bit UUIDs, and return error if they
+ *                          do not fit in @p data_buf.
+ *  - BT_DATA_UUID32_SOME:  Encode 32-bit UUIDs until @p data_buf is full.
+ *  - BT_DATA_UUID32_ALL:   Encode all 32-bit UUIDs, and return error if they
+ *                          do not fit in @p data_buf.
+ *  - BT_DATA_UUID128_SOME: Encode 128-bit UUIDs until @p data_buf is full.
+ *  - BT_DATA_UUID128_ALL:  Encode all 128-bit UUIDs, and return error if they
+ *                          do not fit in @p data_buf.
+ * @param[out] data           Pointer to the bt_data struct to store the
+ *                            results in.
+ * @param[out] data_buf       Pointer to the buffer where the encoded UUID data
+ *                            is stored.
+ * @param[out] data_buf_size  Size of the @p data_buf
+ *
+ * @retval 0 on success.
+ *
+ * @retval -EINVAL if any of the parameters are invalid (e.g. NULL or 0).
+ *
+ * @retval -ENOMEM if a _ALL type is used and @p data_buf does not fit all the
```

Comment:
The implementation leaves the out-parameters as if a _SOME type was used in the case or -ENOMEM. I can see users relying on this undocumented behavior. I'm not sure if it is a problem. (Is it?) We could deal with it now and specify this behavior in the API, or sabotage the output with memset (or at least set `data_len` to 0).

Reply:
Done :)